### ISSUE TITLE
pyzmq version 21.0.1 does not work on mac os High Sierra version 10.13.6

### ISSUE THREAD

**Issue by atwellke** [zeromq/pyzmq#1480]
### Description

When I create a new virtual environment with pipenv, install jupyterlab and then run jupyter notebook I get an ImportError. I'm using Python 3.7.

### Steps to Reproduce

1. Create a new virtual environment (pipenv install)
2. Install JupyterLab (pipenv install jupyterlab)
3. Run Jupyter Notebook (pipenv run jupyter notebook)

### Expected Results

Jupyter notebook should then open in my browser

### Actual Results

``` 
Traceback (most recent call last):
  File "/Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/bin/jupyter-notebook", line 5, in <module>
    from notebook.notebookapp import main
  File "/Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/lib/python3.7/site-packages/notebook/notebookapp.py", line 49, in <module>
    from zmq.eventloop import ioloop
  File "/Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/lib/python3.7/site-packages/zmq/__init__.py", line 62, in <module>
    from zmq import backend
  File "/Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/lib/python3.7/site-packages/zmq/backend/__init__.py", line 32, in <module>
    raise original_error from None
  File "/Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/lib/python3.7/site-packages/zmq/backend/__init__.py", line 27, in <module>
    _ns = select_backend(first)
  File "/Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/lib/python3.7/site-packages/zmq/backend/select.py", line 32, in select_backend
    mod = import_module(name)
  File "/Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/lib/python3.7/site-packages/zmq/backend/cython/__init__.py", line 6, in <module>
    from . import (
ImportError: dlopen(/Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/lib/python3.7/site-packages/zmq/backend/cython/constants.cpython-37m-darwin.so, 2): Symbol not found: ____chkstk_darwin
  Referenced from: /Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/lib/python3.7/site-packages/zmq/backend/cython/../../.dylibs/libsodium.23.dylib (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib
 in /Users/katiea/.local/share/virtualenvs/pyzmq_error-dNZK0c9B/lib/python3.7/site-packages/zmq/backend/cython/../../.dylibs/libsodium.23.dylib 
```

### Current workaround

If I downgrade pyzmq to an earlier version (20.0.0) it works:-

1. Create a new virtual environment (pipenv install)
2. Install earlier version of pyzmq (pipenv install pyzmq==20.0.0)
3. Install JupyterLab (pipenv install jupyterlab)
4. Run Jupyter Notebook (pipenv run jupyter notebook)

This works as expected

**Comment by denis-bz** [comment]
On macosx 10.10 (admittedly old) with pyzmq-20.0.0, I get the same error
`.../zmq/backend/cython/constants.cpython-37m-darwin.so, 2): Symbol not found: ____chkstk_darwin`
19.0.2 seems to work, so far.

**Comment by minrk** [comment]
Thanks for testing! I think I have some missing `MACOSX_DEPLOYMENT_TARGET`s. `pip install --no-binary pyzmq` should let you build from source in the meantime.

mac artifacts for #1484 will include updated mac builds with the change, if you could test those and confirm they fix it. I no longer have access to an old mac OS to test with.

**Comment by minrk** [comment]
@atwellke @denis-bz  if you could test one or more of the wheels-mac artifacts [from this build](https://github.com/zeromq/pyzmq/pull/1484/checks) ([direct download link](https://github.com/zeromq/pyzmq/suites/1893674267/artifacts/37066476)) and report back if this fixes the issue, that would be super helpful. I've confirmed that at least the dylib *claims* to support 10.9 (LC_VERSION_MIN_MACOSX has changed from 10.15 to 10.9 in the dylibs).

**Comment by denis-bz** [comment]
Hi @minrk, `pip install --prefix .../site-packages  pyzmq  --no-binary :all:` works except for
```
[W 13:14:30.606 NotebookApp] 404 GET /nbextensions/widgets/notebook/js/extension.js?v=20210125131426 (::1) 3.210000ms referer=http://localhost:8888/notebooks/ipympl.ipynb
[IPKernelApp] WARNING | Config option `prompts_class` not recognized by `ZMQInteractiveShell`.  Did you mean one of: `prompt_in2, prompt_out, prompts_pad_left`?
```

May I ask you different question, is there a SHORT intro to jupyter notebook for someone who knows IPython fairly well,
10 pages or so ?  The pdf on readthedocs is 176 pages ...
Thanks

**Comment by minrk** [comment]
Thanks, good to know building from source works. Were you able to test the wheels linked above? You can try out demos of various Jupyter things at https://jupyter.org/try .

**Comment by denis-bz** [comment]
@minrk, yes, pyzmq-21.0.2.dev0-cp37-cp37m-macosx_10_9_x86_64.whl works on my macos 10.10
+ fwiw jupyter_client-6.1.11  notebook-6.2.0

**Comment by minrk** [comment]
Perfect, thank you! Since 10.10 is older than 10.13, I'm going to assume it's fixed for you as well @atwellke. But don't hesitate to post here if you find otherwise.

**Comment by denis-bz** [comment]
@minrk, could you answer [cannot-run-jupyter-everywhere-terminal-anaconda-distributer-mac-os-10-12-6](https://stackoverflow.com/questions/65774567/cannot-run-jupyter-everywhere-terminal-anaconda-distributer-mac-os-10-12-6) on SO
and if possible post what you did (`MACOSX_DEPLOYMENT_TARGET`, anything else ?)
a heads-up for other people who build xxx for Macs and can't test on old Macs
(don't know if there's a central wiki where wheel-builders look)
Thanks !

**Comment by minrk** [comment]
answered and released 21.0.2 with the fix
